src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -103,6 +103,7 @@ public TCommandResult Execute(IConnection connection, CancellationToken cancella
                 finally
                 {
                     MessageWasProbablySent(message);
+                    UpdateTransactionStateIfRequired(message);
```

Comment:
Replace lines 105-106 with:

    if (message.WasSent) { MessageWasProbablySent(message); }

Reply:
Or if you prefer this formatting:

    if (message.WasSent)
    {
        MessageWasProbablySent(message);
    }